pages/en/lb3/Using-current-context.md

```diff
@@ -10,156 +10,147 @@ permalink: /doc/en/lb3/Using-current-context.html
-As a result, loopback-context does not work in many situations, as can be seen from issues reported in LoopBack's
-[issue tracker](https://github.com/strongloop/loopback/issues?utf8=%E2%9C%93&q=is%3Aissue%20getCurrentContext).
-
-See [loopback issue #1495](https://github.com/strongloop/loopback/issues/1495) updates and an alternative solution.
-" %}
-
 LoopBack applications sometimes need to access context information to implement the business logic, for example to:
 
 * Access the currently logged-in user.
 * Access the HTTP request (such as URL and headers).
 
 A typical request to invoke a LoopBack model method travels through multiple layers with chains of asynchronous callbacks. It's not always possible to pass all the information through method parameters. 
 
-**See also**: [Example in LoopBack repository](https://github.com/strongloop/loopback/blob/master/example/context/app.js).
+In LoopBack 2.x, we have introduced current-context APIs which used the module
+[continuation-local-storage](https://www.npmjs.com/package/continuation-local-storage)
+to provide a context object preserved across asynchronous operations.
+Unfortunately, as we learned later, this module is not reliable and have many
+problems (for example, see [issue #59](https://github.com/othiym23/node-continuation-local-storage/issues/59)).
+As a result, our current-context feature does not work in many situations,
+as can be seen from issues reported in LoopBack's
+[issue tracker](https://github.com/strongloop/loopback/issues?utf8=%E2%9C%93&q=is%3Aissue%20getCurrentContext).
 
-## Error messages
+To address this problem, we have moved all current-context-related code to a
+standalone module [loopback-context](https://github.com/strongloop/loopback-context)
+and removed all current-context APIs in the version 3.0 (see
+[Release Notes](3.0-Release-Notes.html#current-context-api-and-middleware-removed) for
+more details).
 
-LoopBack 3.0 [removed current-context APIs](3.0-Release-Notes.html#current-context-api-and-middleware-removed).
-An application that uses current-context will print the following error message
-when it receives its first HTTP request:
+Having done that, we do recognize the need of accessing information like the
+currently logged-in user deep inside application logic, for example in
+[Operation Hooks](Operation-hooks.html). Until there is a reliable
+implementation of continuation-local-storage available for Node.js, we are
+recommending to explicitly pass any additional context via `options` parameter
+of (remote) methods.
 
-```
-Unhandled error for request GET /api/Users:
-Error: remoting.context option was removed in version 3.0.
-For more information, see https://loopback.io/doc/en/lb3/Using-current-context.html
-for more details.
-    at restApiHandler (.../node_modules/loopback/server/middleware/rest.js:44:15)
-    at Layer.handle [as handle_request] (.../node_modules/express/lib/router/layer.js:95:5)
-    ...
-```
+All built-in methods like
```

Comment:
Built-in method were modified in both 3.x and 2.x. Should I explicitly mention this fact?

(IIRC, "options" parameter was added before we started this current-context rework.)